packages/@aws-cdk/aws-iotevents/test/input.test.ts

```diff
@@ -68,11 +101,48 @@ test('can import a Input by inputName', () => {
 });
 
 test('cannot be created with an empty array of attributeJsonPaths', () => {
-  const stack = new cdk.Stack();
-
   expect(() => {
     new iotevents.Input(stack, 'MyInput', {
       attributeJsonPaths: [],
     });
   }).toThrow('attributeJsonPaths property cannot be empty');
 });
+
+
+test('can grant the permission to put message', () => {
+  const role = iam.Role.fromRoleArn(stack, 'MyRole', 'arn:aws:iam::account-id:role/role-name');
+  const input = new iotevents.Input(stack, 'MyInput', {
+    inputName: 'test_input',
+    attributeJsonPaths: ['payload.temperature'],
+  });
+
+  // WHEN
+  input.grantPutMessage(role);
+
+  // THEN
+  Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
```

Comment:
Feel like the Role the Policy points to should be asserted here too.

Reply:
I've added the property `Roles: ['role-name'],` to the expected object of this assertion.

Reply:
Nice!